### DISCUSSION TITLE
Outlet json for an automatic turn on the device after 30 seconds

### DISCUSSION THREAD

**Discussion by Hanimaniac** [RavenSystem/esp-homekit-devices#1807] [MEPLHAA Scripts]
Hello Community,

i use this json: [{"c":{"l":13,"b":[{"g":0,"t":5}]},"a":[{"t":2,"i":30,"0":{"r":[{"g":4,"v":1}]},"s":1,"1":{"r":[{"g":4,"v":0}]},"b":[{"g":5,"p":0,"i":1,"t":0}]}]}]() for an outlet.

My Goal is, that the device is always ON, when turning it OFF, it should automatically switch ON after 30 Seconds.
With this json, it works the opposite way.
I hope you can help me.
Greeting
Hanimaniac

Edit 01.04.2022 (no April fool :) )
For better understanding, I use an EnOcean Hub from a german company called SENIC to control my sonos devices with a Wallswitch ( they are the same as friends of hue switches).
After 2-3 Weeks, I can't controll my speakers over the switch, cause the hub isn't responding. So unplug the hub, wait 30 seconds, and plug it in mostly helps. So I'd like to put a little switch between who can do this anointing job for me and thats why I bneed Your help.

Hanimaniac

**Comment by Ucoz3000** [comment]
[{"c":{"l":13,"b":[{"g":0,"t":5}]},"a":[{"t":2,"i":30,"0":{"r":[{"g":4,"v":0}]},"s":1,"1":{"r":[{"g":4,"v":1}]},"b":[{"g":5,"p":0,"i":1,"t":0}]}]}]

**Reply by Hanimaniac** [reply]
Thanks for the Answer, 
but it doesn't work the way I want. 
Also with this Json, when you turn it ON, it turns OFF after 30 seconds.
When you turn it OFF, it stays OFF.

**Reply by xrust83** [reply]
on the phone screen?  but in fact, when the screen is off, it works, and if you turn it on, the relay contacts will turn off.  what looks like it's off is actually working.

**Reply by Hanimaniac** [reply]
Hello xrust83,
Good point, I have to figure that out.
I yust looked on the "Software side" jet.

**Reply by Hanimaniac** [reply]
Hello Xrust83,
I tested it now with Hardware on (Transistor with LED) and it works like "on screen).

**Reply by xrust83** [reply]
Parameter "i": 30 one-sided. If you press the button, step 1 returns to step 0, but not vice versa. Your source code was correct, except for the parameter "c": 1, since you inverted the outputs r: 4 0 = 1 and 1 = 0 and c must be inverted to 0 (ie when the power supply is performed action 0 (g: 12, v: 1)
I changed the json for sonoff basics (button gpio 0, relay gpio 12, and time 10 sec.) I think everything works as it should

{"c":{"l":13,"i":0,"b":[{"g":0,"t":5}]},"a":[{"t":2,"i":10,"0":{"r":[{"g":12,"v":1}]},"s":0,"1":{"r":[{"g":12,"v":0}]},"b":[{"g":0,"p":0,"i":1,"t":0}]}]}

{"c":{"l":13,"i":0,"b":[{"g":0,"t":5}]},"a":[{"t":2,"i":10,"0":{"r":[{"g":12,"v":1}]},"1":{"r":[{"g":12}]},"b":[{"g":0,"p":0,"i":1,"t":0}]}]}

for you

{"c":{"l":13,"i":0,"b":[{"g":0,"t":5}]},"a":[{"t":2,"i":30,"0":{"r":[{"g":4,"v":1}]},"1":{"r":[{"g":4}]},"b":[{"g":5,"p":0,"i":1,"t":0}]}]}

**Reply by xrust83** [reply]
https://www.youtube.com/watch?v=8nEGMP-vCL8

**Reply by xrust83** [reply]
It looks like a solution for an electromagnetic lock, constantly keeps the door closed but when you press the button turns off for 30 seconds giving the opportunity to open the door giving the opportunity to pass :-)

**Reply by Hanimaniac** [reply]
Xrust83 ❤️
What should I say, I tested it and it works. Thank you verymuch for your time and help (and the video ofcourse). 
One thing, the json for me is invalid, just change "g"5 to "g":5 (if anyone else uses this json). 
Thanks and have a nice day
Hanimaniac

**Reply by xrust83** [reply]
yes, it's a typo =), thanks already fixed
